src/web/components/SignInGate/SignInGateSelector.tsx

```diff
@@ -116,40 +116,42 @@ export const SignInGateSelector = ({
 	isSignedIn,
 	CAPI,
 }: SignInGateSelectorProps) => {
-	const [showGate, setShowGate] = useState(true);
+	const [isGateDismissed, setIsGateDismissed] = useState(false);
+	const [canShowGate, setCanShowGate] = useState(false);
 	const gateSelector = useSignInGateSelector();
 
 	useEffect(() => {
```

Comment:
Is the use case here that this event only fires once? If so, maybe DCR's custom `useOnce` hook would be useful?

Reply:
ah interesting! i'll take a look at the `useOnce` hook

Reply:
The thing to know about `useOnce` is it works slightly differently to `useEffect`. In useEffect, if any of the dependencies _change_ then this triggers the code to re-execute. In useOnce the code is _only_ executed once all the dependencies are truthy.

I think in this case you can probably swap it in directly and it should work. You'll also be able to delete the `if (isGateDismissed) {` check as this is implicit

Reply:
Great, I've changed it to use a `useOnce` hook